android/src/main/java/com/zmxv/RNSound/RNSoundModule.java

```diff
@@ -56,6 +66,19 @@ protected MediaPlayer createMediaPlayer(final String fileName) {
     if (res != 0) {
       return MediaPlayer.create(this.context, res);
     }
+    if(fileName.startsWith("http://") || fileName.startsWith("https://")) {
+      MediaPlayer mediaPlayer = new MediaPlayer();
+      mediaPlayer.setAudioStreamType(AudioManager.STREAM_MUSIC);
+      Log.i("RNSoundModule", fileName);
+      try {
+        mediaPlayer.setDataSource(fileName);
+      } catch(IOException e) {
+        Log.e("RNSoundModule", "Exception", e);
+        return null;
```

Comment:
Did you have a specific reason not to invoke a callback error when the setDataSource throws an exception?